src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1223 @@
+            }
+        }
+
+        return subFeature;
+    }
+
+    /**
+     * Creates a {@link org.broadinstitute.hellbender.tools.funcotator.FuncotatorUtils.SequenceComparison} object with the fields populated.
+     * @param variant The {@link VariantContext} for the current variant.
+     * @param alternateAllele The current alternate {@link Allele} for the variant.
+     * @param reference The {@link ReferenceContext} for the current sample set.
+     * @param transcript The {@link GencodeGtfTranscriptFeature} for the current gene feature / alt allele.
+     * @param exonPositionList A {@link List} of {@link htsjdk.samtools.util.Locatable} objects representing exon positions in the transcript.
+     * @return A populated {@link org.broadinstitute.hellbender.tools.funcotator.FuncotatorUtils.SequenceComparison} object.
+     */
+    @VisibleForTesting
+    static FuncotatorUtils.SequenceComparison createSequenceComparison(final VariantContext variant,
+                                                                final Allele alternateAllele,
+                                                                final ReferenceContext reference,
+                                                                final GencodeGtfTranscriptFeature transcript,
+                                                                final List<? extends htsjdk.samtools.util.Locatable> exonPositionList,
+                                                                final Map<String, MappedTranscriptIdInfo> transcriptIdMap,
+                                                                final ReferenceDataSource transcriptFastaReferenceDataSource) {
+
+        final FuncotatorUtils.SequenceComparison sequenceComparison = new FuncotatorUtils.SequenceComparison();
+
+        // Get the contig:
+        sequenceComparison.setContig(variant.getContig());
+
+        // Get the strand:
+        final Strand strand = Strand.decode( transcript.getGenomicStrand().toString() );
+        sequenceComparison.setStrand(strand);
+
+        // Get the alleles from the inputs
+        // Also get the reference sequence for the variant region
+        // (spanning the entire length of both the reference and the variant, regardless of which is longer).
+        final Allele refAllele;
+        final Allele altAllele;
+
+        // TODO: Make this a parameter:
+        final int referenceWindow = 10;
+        final String referenceBases;
+
+        if ( strand == Strand.POSITIVE ) {
+            refAllele = variant.getReference();
+            altAllele = alternateAllele;
+
+            // Calculate our window to include any extra bases but also have the right referenceWindow:
+            final int endWindow = refAllele.length() >= altAllele.length() ? referenceWindow + refAllele.length() - 1: referenceWindow + altAllele.length() - 1;
+
+            // Set our reference window:
+            reference.setWindow(referenceWindow, endWindow);
```

Comment:
Can you have a look at all the places where you call `setWindow()` on a `ReferenceContext`, and check whether you need to restore the previous window when done? Actually, what would be even better if you have time, would be to add a new `getBases()` method to `ReferenceContext` that takes a window as an extra argument and doesn't permanently change the window stored in the `ReferenceContext` itself, and use that overload everywhere.

Reply:
Added.  I'll go through and check where I'm grabbing bases and use this instead.